resources/qml/Preferences/ProfilesPage.qml

```diff
@@ -1,355 +1,531 @@
+        }
+    }
 
-            ProfileTab
-            {
-                title: catalog.i18nc("@title:tab", "Global Settings");
-                quality: base.currentItem != null ? base.currentItem.id : "";
-                material: Cura.MachineManager.allActiveMaterialIds[Cura.MachineManager.activeMachineId]
-            }
+    // Confirmation dialog for removing a profile
+    MessageDialog
+    {
+        id: confirmRemoveQualityDialog
 
-            Repeater
-            {
-                model: base.extrudersModel
+        icon: StandardIcon.Question;
+        title: catalog.i18nc("@title:window", "Confirm Remove")
+        text: catalog.i18nc("@label (%1 is object name)", "Are you sure you wish to remove %1? This cannot be undone!").arg(base.currentItem.name)
+        standardButtons: StandardButton.Yes | StandardButton.No
+        modality: Qt.ApplicationModal
 
-                ProfileTab
-                {
-                    title: model.name;
-                    extruderId: model.id;
-                    extruderDefinition: model.definition;
-                    quality: base.currentItem != null ? base.currentItem.id : "";
-                    material: Cura.MachineManager.allActiveMaterialIds[model.id]
-                }
-            }
+        onYes:
+        {
+            Cura.ContainerManager.removeQualityChangesGroup(base.currentItem.quality_changes_group);
+            // reset current item to the first if available
+            qualityListView.currentIndex = -1;  // TODO: Reset selection.
         }
     }
 
-    Item
+    // Dialog to rename a quality profile
+    UM.RenameDialog
     {
-        UM.I18nCatalog { id: catalog; name: "cura"; }
-
-        UM.ConfirmRemoveDialog
+        id: renameQualityDialog
+        title: catalog.i18nc("@title:window", "Rename Profile")
+        object: "<new name>"
+        onAccepted:
         {
-            id: confirmDialog
-            object: base.currentItem != null ? base.currentItem.name : ""
-            onYes:
-            {
-                var name = base.currentItem.name;
-                Cura.ContainerManager.removeQualityChanges(name)
-                if(Cura.MachineManager.activeQualityName == name)
-                {
-                    Cura.MachineManager.setActiveQuality(base.model.getItem(0).name)
-                }
-                objectList.currentIndex = -1 //Reset selection.
-            }
+            var actualNewName = Cura.ContainerManager.renameQualityChangesGroup(base.currentItem.quality_changes_group, newName);
+            base.newQualityNameToSelect = actualNewName;  // Select the new name after the model gets updated
         }
+    }
 
-        UM.RenameDialog
+    // Dialog for importing a quality profile
+    FileDialog
+    {
+        id: importDialog
+        title: catalog.i18nc("@title:window", "Import Profile")
+        selectExisting: true
+        nameFilters: qualitiesModel.getFileNameFilters("profile_reader")  // TODO: make this easier
```

Comment:
Seems fine to me. Maybe the TODO is unneccessary?

Reply:
Removed outdated TODOs in 6a14e3961